doc/developer-guide/deployment/server-configuration.rst

```diff
@@ -0,0 +1,92 @@
+.. highlight:: bash
+
+.. _guide-deployment-server-configuration:
+
+Server configuration
+====================
+
+This chapter describes how to configure your Linux server for running Zotonic.
+
+.. _file-descriptors:
+
+File descriptors
+----------------
+
+By default, Linux limits the number of file descriptors (i.e. the maximum number
+of open files) at 1024. Running Zotonic at scale will require many more than
+that, particularly because Zotonic uses
+:ref:`WebSockets <controller-websocket>` extensively; remember that every open
+port is an open file.
+
+The limit applies on several levels:
+
+1. the process, in our case the Erlang VM (BEAM) that runs Zotonic
+2. the Zotonic user
+3. system-wide.
+
+To check the current process usage, find BEAM’s PID::
+
+    # pidof beam.smp
+    10006 (for instance)
+
+Then count its open files::
+
+    # lsof -a -p 10006 | wc -l
+
+And compare it to the process limit::
+
+    # cat /proc/10006/limits | grep 'Max open files'
+
+You can raise the process limit by adding ``ulimit -n`` to your Zotonic init
+script::
+
+    ulimit -n 50000
+
+Or change the limit in your system definition. For instance, when using
+systemd::
+
+    [Service]
+    LimitNOFILE=50000
+
+Finally, make sure to check `your system-wide limits`_, too.
+
+Network connections
+-------------------
+
+If you have stateful connection tracking enabled, high-traffic Zotonic sites
+may `overflow your conntrack table`_.
+
+Compare the current number of connections with the limit::
+
+    # sysctl net.netfilter.nf_conntrack_max
+    # sysctl net.netfilter.nf_conntrack_count
+
+If you need to raise the limit, edit ``/etc/sysctl.conf``:
+
+.. code-block:: none
+    :caption: /etc/sysctl.conf
+
+    net.netfilter.nf_conntrack_max = some_number
+
+Moreover, to reduce the number of open connections, you can decrease their
+time-out values so get closed sooner. By default connection stay in the
+conntrack table for 5(!) days. To change this:
```

Comment:
Thanks for the clarification. We sent ping/pong messages over the websocket to check if it is alive...